srv/salt/ceph/upgrade/ses7/adopt.sls

```diff
@@ -0,0 +1,58 @@
+{% if grains.get('oscodename', '') != 'SUSE Linux Enterprise Server 15 SP2' %}
+
+not running sle 15 sp2:
+  test.fail_without_changes:
+    - name: |
+        This host is not running SUSE Linux Enterprise Server 15 SP2.
+        Please upgrade the operating system first, before running the adopt process.
+    - failhard: True
+
+{% endif %}
+
+# This check is so we bail out in case of a half upgraded system (for example,
+# if someone runs `zypper dup` with both SES6 and SES7 repos available, they'll
+# potentially still have ceph nautilus installed, which we really don't want).
+# TODO: figure out how to print a friendly error message in this case
+ceph must be octopus if installed:
+  cmd.run:
+    - name: "[ ! -x /usr/bin/ceph ] || ceph --version | grep -q 'ceph version 15'"
+    - failhard: True
+
+{% set ses7_container_image = salt['pillar.get']('ses7_container_image', 'registry.suse.com/ses/7/ceph/ceph') %}
+
+cephadm:
+  pkg.installed:
+    - name: cephadm
```

Comment:
I had to install `podman` manually, so I think adding it here would be good.  If the install fails because only the SLE15SP2 repos have been added and the SES7 haven't, then the user has to investigate anyway.

Reply:
Done.